scipy/stats/_qmc.py

```diff
@@ -0,0 +1,1493 @@
+              509, 521, 523, 541, 547, 557, 563, 569, 571, 577, 587, 593, 599,
+              601, 607, 613, 617, 619, 631, 641, 643, 647, 653, 659, 661, 673,
+              677, 683, 691, 701, 709, 719, 727, 733, 739, 743, 751, 757, 761,
+              769, 773, 787, 797, 809, 811, 821, 823, 827, 829, 839, 853, 857,
+              859, 863, 877, 881, 883, 887, 907, 911, 919, 929, 937, 941, 947,
+              953, 967, 971, 977, 983, 991, 997][:n]
+
+    if len(primes) < n:
+        big_number = 2000
+        while 'Not enough primes':
+            primes = primes_from_2_to(big_number)[:n]
+            if len(primes) == n:
+                break
+            big_number += 1000
+
+    return primes
+
+
+def van_der_corput(n_samples, base=2, start_index=0, scramble=False,
+                   seed=None):
+    """Van der Corput sequence.
+
+    Pseudo-random number generator based on a b-adic expansion.
+
+    Scrambling uses permutations of the remainders (see [1]_). Multiple
+    permutations are applied to construct a point. The sequence of
+    permutations has to be the same for all points of the sequence.
+
+    Parameters
+    ----------
+    n_samples : int
+        Number of element of the sequence.
+    base : int
+        Base of the sequence.
+    start_index : int
+        Index to start the sequence from.
+    scramble: bool, optional
+        If True, use Owen scrambling.
+    seed : {int or `numpy.random.RandomState` instance}, optional
+        If `seed` is not specified the `numpy.random.RandomState`
+        singleton is used.
+        If `seed` is an int, a new ``RandomState`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``RandomState`` instance, then that
+        instance is used.
+
+    Returns
+    -------
+    sequence : list (n_samples,)
+        Sequence of Van der Corput.
+
+    References
+    ----------
+    .. [1] A. B. Owen. "A randomized Halton algorithm in R",
+       arXiv:1706.02808, 2017.
+
+    """
+    rng = check_random_state(seed)
+    sequence = np.zeros(n_samples)
+
+    quotient = np.arange(start_index, start_index + n_samples)
+    b2r = 1 / base
+
+    while (1 - b2r) < 1:
+        remainder = quotient % base
+
+        if scramble:
+            # permutation must be the same for all points of the sequence
+            perm = rng.permutation(base)
+            remainder = perm[np.array(remainder).astype(int)]
+
+        sequence += remainder * b2r
+        b2r /= base
+        quotient = (quotient - remainder) / base
+
+    return sequence
+
+
+class QMCEngine(ABC):
+    """A generic Quasi-Monte Carlo sampler class meant for subclassing.
+
+    QMCEngine is a base class to construct specific Quasi-Monte Carlo sampler.
+    It cannot be used directly as a sampler.
+
+    Parameters
+    ----------
+    dim : int
+        Dimension of the parameter space.
+    seed : {int or `numpy.random.RandomState` instance}, optional
```

Comment:
This should work with Numpy's new `Generator`s, too, and it doesn't seem to.

```python3
runfile('C:/Users/matth/Dropbox/Scipy/QMC/sphere.py', wdir='C:/Users/matth/Dropbox/Scipy/QMC', current_namespace=True)
Traceback (most recent call last):

  File "C:\Users\matth\Dropbox\Scipy\QMC\sphere.py", line 13, in <module>
    sobol = qmc.Sobol(2, seed=default_rng(0))

  File "C:\Users\matth\scipy\scipy\stats\_qmc.py", line 1179, in __init__
    self._scramble()

  File "C:\Users\matth\scipy\scipy\stats\_qmc.py", line 1188, in _scramble
    self.rng.randint(2, size=(self.dim, self.MAXBIT)),

AttributeError: 'numpy.random._generator.Generator' object has no attribute 'randint'
```

Reply:
If I recall, when we started this we said we did not want to do that (@bashtage). So if we have to now, let me know.

Reply:
Are there difficulties? If it's substantial additional work, I'm not sure if it's needed. But if it's a simple fix, I'd recommend it.

Reply:
I did the following as recommended in numpy's guide:

```
try:
    rg_integers = rg.integers
except AttributeError:
    rg_integers = rg.randint
```

Reply:
Can you remind me what the consensus was on this - is this _only_ going to work with Generators and not RandomState?